### PR TITLE
[FLINK-9550][DOC]FlinkCEP snippet example has some syntax errors

### PR DESCRIPTION
**Pull request by medcv**
## What is the purpose of the change

Fixing FlinkCEP snippet code syntax errors and data type mismatches 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/dev/libs/cep.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ val result: DataStream[Alert] = patternStream.select(createAlert(_))
 The pattern API allows you to define complex pattern sequences that you want to extract from your input stream.
 
 Each complex pattern sequence consists of multiple simple patterns, i.e. patterns looking for individual events with the same properties. From now on, we will call these simple patterns **patterns**, and the final complex pattern sequence we are searching for in the stream, the **pattern sequence**. You can see a pattern sequence as a graph of such patterns, where transitions from one pattern to the next occur based on user-specified
-*conditions*, e.g. `event.getName().equals("start")`. A **match** is a sequence of input events which visits all
```
Comment (dawidwys) [inline docs/dev/libs/cep.md]: Is this change necessary?

File: docs/dev/libs/cep.md [hunk 2]
```diff
@@ -275,36 +275,38 @@ with "foo", and if the sum of the prices of the previously accepted events for t
...
-    public boolean filter(SubEvent value, Context<SubEvent> ctx) throws Exception {
-        if (!value.getName().startsWith("foo")) {
-            return false;
-        }
-
-        double sum = value.getPrice();
-        for (Event event : ctx.getEventsForPattern("middle")) {
-            sum += event.getPrice();
+middle.oneOrMore()
+    .subtype(SubEvent.class)
```
Comment (dawidwys) [inline docs/dev/libs/cep.md]: Either don't add the `subtype` clause or add it also in the scala example.

File: docs/dev/libs/cep.md (hunk 1 above)
Comment (medcv) [reply docs/dev/libs/cep.md]: just to make it consistent with the `return event.getName().equals("end");` in the above snippet

File: docs/dev/libs/cep.md (hunk 2 above)
Comment (medcv) [reply docs/dev/libs/cep.md]: will update!

**Comment by medcv** [comment]
@dawidwys Thanks for the review!

**Comment by medcv** [comment]
@dawidwys PR's been updated! Please review.

**Comment by dawidwys** [comment]
+1, merging
